pkg/metadata/inventories_collector.go

```diff
@@ -75,7 +79,11 @@ func SetupInventories(sc *Scheduler, ac inventories.AutoConfigInterface, coll in
 	}
 	RegisterCollector("inventories", ic)
 
-	if err := sc.AddCollector("inventories", time.Duration(config.Datadog.GetInt("inventories_max_interval"))*time.Second); err != nil {
+	maxInterval := time.Duration(config.Datadog.GetInt("inventories_max_interval")) * time.Second
+	if maxInterval == 0 {
+		maxInterval = config.DefaultInventoriesMaxInterval
+	}
+	if err := sc.AddCollector("inventories", maxInterval); err != nil {
```

Comment:
These seem like something that we could write tests around even if there's no `inventories_collector_test` right now.

Reply:
That will require quite a bit more refactoring to make this testable -- likely introducing more bugs than it solves.  WDYT?

Reply:
I think you can isolate just the config value fetching in each case to a function that could be relatively easy to test. I don't think testing the rest of the file is in scope so no need to cover the other code as part of this.